gem/lib/earlgrey/configure_earlgrey.rb

```diff
@@ -108,243 +128,231 @@ def set_defaults(installer, project_name, test_target_name, scheme_file, opts =
 
     # Main entry point. Configures An Xcode project for use with EarlGrey.
     #
-    # @param installer [Installer] Cococapods installer object or nil.
-    # @param project_name [String] the xcodeproj file name
-    # @param test_target_name [String] the test target name contained in xcodeproj
-    # @param scheme_file [String] the scheme file name. defaults to project name when nil.
+    # @param [String] project_name
+    #        the xcodeproj file name
+    # @param [String] test_target_name
+    #        the test target name contained in xcodeproj
+    # @param [String] scheme_file
+    #        the scheme file name. defaults to project name when nil.
     # @return [nil]
-    def configure_for_earlgrey(installer, project_name, test_target_name, scheme_file, opts = {})
-      set_defaults(installer, project_name, test_target_name, scheme_file, opts)
+    def configure_for_earlgrey(project_name, test_target_name,
+                               scheme_file, opts = {})
+      set_defaults(project_name, test_target_name, scheme_file, opts)
 
-      # Add a Test Action to the User Project Scheme.
-      scheme = modify_scheme_for_actions
+      # Add DYLD_INSERT_LIBRARIES to the schemes
+      modify_scheme_for_actions(user_project, [test_target]).each do |_, scheme|
+        scheme.save!
+      end
 
-      # Add a Copy Files Build Phase for EarlGrey.framework to embed it into the app under test.
-      add_earlgrey_copy_files_script
+      # Add a Copy Files Build Phase for EarlGrey.framework to embed it into
+      # the app under test. Carthage uses carthage copy-frameworks instead of
+      # a copy files build phase.
+      framework_ref = add_earlgrey_product user_project, carthage
```

Comment:
```ruby
# Carthage uses carthage copy-frameworks instead of
# a copy files build phase.
```
^ that should be deleted, we're now using the same `Copy Files Build Phase` for both cocoapods and carthage.